digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/OracleAssetDao.scala

```diff
@@ -127,4 +127,24 @@ class OracleAssetDao {
     Queries.updateAssetModified(id, username).first
     sqlu"update asset set valid_to = sysdate where id = $id".first
   }
+
+  def getExpiredAssetsByRoadLinkAndTypeIdAndBetweenDates(roadLinksFromMunicipalities: Set[Long], assetTypeIdsToRestore: Seq[Int], startDate: String, endDate: String) = {
+    MassQuery.withIds(roadLinksFromMunicipalities) { idTableName =>
+      sql"""
+            select a.id
+            from asset a
+            join asset_link al on al.asset_id = a.id
+            join lrm_position lp on lp.id = al.position_id
+            join #$idTableName i on i.id = lp.link_id
+            where a.asset_type_id in (#${assetTypeIdsToRestore.mkString(",")}) and (a.valid_to between to_date($startDate, 'yyyy-mm-dd hh24:mi:ss') and to_date($endDate, 'yyyy-mm-dd hh24:mi:ss'))
+         """.as[Long].list
+    }
+  }
+
+  def restoreExpiredAssetsByIds(assetsToBeRestored: Set[Long], username: String) = {
+    sqlu"""
+          update asset
+          set modified_date = sysdate, modified_by = $username, valid_to = null
+          where id in (#${assetsToBeRestored.mkString(",")})""".execute
```

Comment:
Can you guarantee that the assetsToBeRestored is maximum of 1000? Because if not it will fail since the in operand can only contain 1000 cases.

Reply:
Applied